### PR TITLE
TableReader returns ArrowSchema for request

### PR DESCRIPTION
**Pull request by metalmatze**
It's helpful to return the computed schema for a request. 
The `TableReader` now has an `ArrowSchema` method that under the hood uses the same `ParquetRowGroupToArrowRecord` function that's used to generate the Arrow Records that now additionally returns the schema belonging to those records.

### CODE REVIEW

File: pqarrow/arrow.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func ParquetRowGroupToArrowRecord(
 	projections []logicalplan.ColumnMatcher,
 	filterExpr logicalplan.Expr,
 	distinctColumns []logicalplan.ColumnMatcher,
-) (arrow.Record, error) {
+) (arrow.Record, *arrow.Schema, error) {
```
Comment (brancz) [inline pqarrow/arrow.go:28]: Am I missing something or can we already read the schema from the `arrow.Record`? If so I think we should just return the `arrow.Record` by itself and let the caller decide what to do.

File: table.go [hunk 2]
```diff
@@ -949,7 +1010,7 @@ func (t *TableBlock) Serialize() ([]byte, error) {
 
 	// Read all row groups
 	rowGroups := []dynparquet.DynamicRowGroup{}
-	err := t.RowGroupIterator(ctx, nil, &AlwaysTrueFilter{}, true, func(rg dynparquet.DynamicRowGroup) bool {
+	err := t.RowGroupIterator(ctx, math.MaxUint64, nil, &AlwaysTrueFilter{}, func(rg dynparquet.DynamicRowGroup) bool {
```
Comment (brancz) [inline table.go:1013]: maybe put a comment on the use of `math.MaxUint64` here

**Review by brancz — COMMENTED** [review]
This looks great! I do wonder whether at this point introducing a first call `Txn` struct would make sense, but I don't feel strongly about introducing it now. It feels inevitable though.

**Comment by metalmatze** [comment]
> This looks great! I do wonder whether at this point introducing a first call `Txn` struct would make sense, but I don't feel strongly about introducing it now. It feels inevitable though.

Agreed, I was thinking the same, for now, it should be ok and we can improve down the road.

**Comment by brancz** [comment]
lgtm 👍 🎉

**Comment by brancz** [comment]
Nice work!!

**Pull request merged**